src/ring_theory/localization.lean

```diff
@@ -236,6 +236,33 @@ lemma exist_integer_multiples_of_finset (s : finset S) :
   ∃ (b : M), ∀ a ∈ s, is_integer R ((b : R) • a) :=
 exist_integer_multiples M s id
 
+/-- A choice of a common multiple of a finite set of fractions. -/
+noncomputable
+def denominator_common_multiple (s : finset S) : M :=
+(exist_integer_multiples_of_finset M s).some
```

Comment:
Have you tried basing this on `exist_integer_multiples` instead? That would be more generally useful, for example for the coefficients of a polynomial `p` you can do something like `denominator_common_multiple p.support p.coeff` (rather than `denominator_common_multiple (p.support.image p.coeff)` followed by rewriting with `finset.mem_image`).

Reply:
This is used for showing things like finitely-generated-ness where I had to deal with `finset R`.
So I think I would need still need the definitions for `finest` explicitly or else I would have to deal with `finest.image` and `finset.attach`.

I could still give a variant using `exist_integer_multiples` and base my definitions on it.